docs/authentication.md

```diff
@@ -0,0 +1,128 @@
+# A Guide on ScalarDL Authentication
+
+This document explains the ScalarDL authentication mechanism and how to use it properly. 
+
+## Authentication in ScalarDL
+
+Authentication is one of the key roles in ScalarDL to make the protocol work as expected.
+ScalarDL uses authentication for the following three places:
+* Client authentication (for Ledger and Auditor)
+    * Client authentication authenticates clients by Ledger and Auditor using client-generated signatures attached to requests from clients.
+* Ledger authentication (for Auditor)
+    * Ledger authentication authenticates Ledger by Auditor using Ledger-generated signatures attached to [Asset Proofs](how-to-use-proof.md).
+* Auditor authentication (for Ledger )
+    * Auditor authentication authenticates Auditor by Ledger using  Auditor-generated signatures attached to client requests.
+
+Note that Ledger authentication and Auditor authentication are only used in the Auditor mode. For more details about Auditor, please see the [Getting Started with ScalarDL Auditor](getting-started-auditor.md) and [ScalarDL Implementation](implementation.md).
+
+## Authentication methods
+
+ScalarDL supports two authentication methods: Digital Signatures and HMAC.
```

Comment:
I guess it might be more precise to revise the description above

For example,
Revise 
```
Ledger and Auditor use the client-generated signatures attached to request to authenticate clients
```
to be
```
Ledger and Auditor use the client-generated signatures or message codes attached to request to authenticate clients
```

because we have two possible authentication methods here.

Reply:
Thank you for the comment. It's a good point.
Here, we use `signature` to mean digital signatures and MAC.
As you sayd, when we say `signature`, it used to be `digital signature`, but lately it's been used for (H)MAC as wel., and the ScalarDL code uses `signature` field to store digital signatures and HMAC, so I think it's more consistent to use `signature` here.